src/dmd/json.d

```diff
@@ -857,17 +858,165 @@ public:
 extern (C++) void json_generate(OutBuffer* buf, Modules* modules)
 {
     scope ToJsonVisitor json = new ToJsonVisitor(buf);
-    json.arrayStart();
-    json.generateCompilerInfo();
-    json.generateBuildInfo();
-    for (size_t i = 0; i < modules.dim; i++)
-    {
-        Module m = (*modules)[i];
-        if (global.params.verbose)
-            message("json gen %s", m.toChars());
-        m.accept(json);
-    }
-    json.generateSemantics();
-    json.arrayEnd();
-    json.removeComma();
+
+    if (global.params.jsonQuery is null)
+    {
+        // Generate the original "non-query format, which is just an array
+        // of modules representing their syntax.
+
+        json.arrayStart();
+        for (size_t i = 0; i < modules.dim; i++)
+        {
+            Module m = (*modules)[i];
+            if (global.params.verbose)
+                message("json gen %s", m.toChars());
+            m.accept(json);
+        }
+        json.arrayEnd();
+        json.removeComma();
+    }
+    else
+    {
+        // Generate the "query format" which is an object where each
+        // output option is it's own field.
+
+        json.objectStart();
+        auto parser = QueryParser(global.params.jsonQuery);
+        for (;;)
```

Comment:
`while(true)` seems more idiomatic?

Reply:
FYI: `for (;;)` is the "idiomatic" way of Phobos. We should probably put such trivias in the style guide, s.t. they don't constantly pop up...